assets/js/components/Dashboard/Dashboard.tsx

```diff
@@ -25,7 +42,8 @@ const Dashboard = (props: { page: string }): JSX.Element => {
   );
   const [statusFilterValue, setStatusFilterValue] = useState(STATUSES[0]);
   const [activeEventKeys, setActiveEventKeys] = useState<string[]>([]);
-  const [sortLabel, setSortLabel] = useState("ABC");
+
+  const sortLabel = getSortLabel(modeLineFilterValue, sortDirection);
```

Comment:
This is more of a React question than a suggestion. Is there a particular advantage to passing state values as parameters? Is that just to allow for calling the function with values that are not in state?

Reply:
It's mostly 6 and 1/2 dozen, but there are a few reasons I prefer to do it this way when possible:
- it forces you to explicitly state in the argument list what variables the function depends on, as opposed to a [closure](https://en.wikipedia.org/wiki/Closure_(computer_programming)) that uses one or more free variables from its containing scope. You get a pure function: variable(s) in -> value out.
- (much less important) a slight performance gain. See below 👇

Technically, the entire body of a component runs anew every time it re-renders, so when you define a helper function inside of the component, that definition is recreated on every re-render, as opposed to a function defined outside at the module level which is defined only once. Overall it doesn't make a huge difference performance-wise, though. Function definitions are fast.